src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -474,16 +471,18 @@ class NetworkOPsImp final
     void onStop () override
     {
         mAcquiringLedger.reset();
-        m_heartbeatTimer.cancel();
-        m_clusterTimer.cancel();
+
+        if (heartbeatTimer_)
+            heartbeatTimer_->cancel();
+        if (clusterTimer_)
+            clusterTimer_->cancel();
 
         stopped ();
```

Comment:
There's no explicit way to know, you just have to structure code so that there is no race. @nbougalis ?